### ISSUE TITLE
1.20.x/1.21.x compatibility

### ISSUE THREAD

**Issue by SlimeDog** [gamerover98/NoBlackSky#6]
Paper 1.20.2
NoBlackSky 2.10

On server start-up
```
[11:10:14] [Server thread/INFO]: [NoBlackSky] Loading server plugin NoBlackSky v2.10
[11:10:14] [Server thread/WARN]: [NoBlackSky] §eCannot find the current server version, attempting to start the plugin with the latest (1.20.1) supported version ...
[11:10:50] [Server thread/INFO]: [NoBlackSky] Enabling NoBlackSky v2.10
```

**Comment by gamerover98** [comment]
Already fixed here: d19c953d06f077c86ad2a34c87dbb41cce4f430d.
I will release the compiled jar soon.

**Comment by gamerover98** [comment]
I'm forced to reopen this issue because it seems that the [play login packet](https://wiki.vg/Protocol#Login_.28play.29) and the [play respawn packet](https://wiki.vg/Protocol#Respawn) don't contain the world name, unlike 1.20.1 and older versions.

Is this an issue with ProtocolLib? I'm using the latest available build: [#669](https://ci.dmulloy2.net/job/ProtocolLib/669/)

----------------
**Edit**: I have opened an issue on https://github.com/dmulloy2/ProtocolLib/issues/2569

**Comment by SlimeDog** [comment]
Any progress on this issue?

**Comment by gamerover98** [comment]
Nope, the [issue](https://github.com/dmulloy2/ProtocolLib/issues/2569) is still open.

**Comment by SlimeDog** [comment]
As you note, the API has changed. I think there are different API calls to get what you need.

**Comment by gamerover98** [comment]
The API is not changed, but the result does.
_I realized that I made a mistake on the ProtocolLib issue description and I fixed it_.

What I expect for 1.20.2 is the same behavior as 1.20.1. I already checked the [login](https://wiki.vg/Protocol#Login_.28play.29) and the [respawn](https://wiki.vg/Protocol#Respawn) packets and the `Dimension Name` property is still there. I'll examine the ProtocolLib source to find a solution...

**Comment by SlimeDog** [comment]
> The API is not changed, but the result does. 

The result is part and parcel of the API. The result changed, as you noted.
 
> What I expect for 1.20.2 is the same behavior as 1.20.1.

As I understand it, that is what changed. You will have to use different API calls to get the data you need.

**Comment by gamerover98** [comment]
> You will have to use different API calls to get the data you need.

Okay, but I would appreciate it if you could give me a hand 👍

**Comment by SlimeDog** [comment]
Sorry, I don't use the API. You can search the documentation (or ask for help in forums) as fast as I can.

**Comment by skbeh** [comment]
@gamerover98 There may be a solution: https://github.com/dmulloy2/ProtocolLib/issues/2571#issuecomment-1762780495.

**Comment by SlimeDog** [comment]
There have been ProtocolLib releases since this issue was posted (latest build 679). Does it give the desired result now?

**Comment by SlimeDog** [comment]
Issue remains with
Paper 1.20.6-137 (or previous)
ProtocolLib 5.3.0-SNAPSHOT-699 (or previous)
NoBlackSky 2.10

**Comment by SlimeDog** [comment]
Issue remains with
Paper 1.21-9 (or previous)
ProtocolLib 5.3.0-SNAPSHOT-720 (or previous)
NoBlackSky 2.10

**Comment by SlimeDog** [comment]
Issue remains with
Paper 1.21.3-25 (or previous)
ProtocolLib 5.4.0-SNAPSHOT-735 (or previous)
NoBlackSky 2.10

**Comment by LucidAPs** [comment]
I tried to tackle this issue myself but this is as far I could go, to me it seems that due to changes in 21 java and latest paper, reflection has been restricted massively, and protocollib unfortunately doesn't really help to make things clearer and accessible. It seems the only solution is to dive into NMMs (good luck with that). I will leave my code and findings here.

My code.
https://paste.ofcode.org/WTFwsJhPasAvQY5kFT7Cus

Error result in game by the above code.
https://paste.ofcode.org/r8cEWmn2YtPfc6kQVpN5qA

**Comment by SlimeDog** [comment]
Effectively abandoned, and I no longer care. Closing...
